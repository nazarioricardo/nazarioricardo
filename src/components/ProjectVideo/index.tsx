import ReactPlayer from "react-player";
import styles from "./styles.module.css";

type ProjectVideoProps = {
  value: { url: string };
};

function ProjectVideo({ value: { url } }: ProjectVideoProps) {
  return (
    <div className={styles.postVideo}>
      <ReactPlayer url={url} />
    </div>
  );
}

export default ProjectVideo;
