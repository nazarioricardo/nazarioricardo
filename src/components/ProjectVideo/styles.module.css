.postVideo {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  text-align: center;
  border-radius: 50px;
  overflow: hidden;
}
