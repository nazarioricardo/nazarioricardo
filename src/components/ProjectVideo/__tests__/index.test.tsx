import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ProjectVideo from "../index";
import { act } from "react-dom/test-utils";

describe("ProjectVideo", () => {
  const mockProps = {
    value: {
      url: "https://test-video-url.com",
    },
  };

  test("renders ProjectVideo component", async () => {
    render(<ProjectVideo {...mockProps} />);

    let videoContainerElement: HTMLElement = await act(() => {
      return screen.getAllByRole("generic")[2];
    });

    expect(videoContainerElement).toBeInTheDocument();
    expect(videoContainerElement).toHaveAttribute(
      "style",
      "border-radius: 50px; overflow: hidden; width: 640px; height: 360px;"
    );

    const videoElement = videoContainerElement.firstChild;
    expect(videoElement).toHaveAttribute("src", mockProps.value.url);
  });

  test("matches snapshot", () => {
    const { asFragment } = render(<ProjectVideo {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
