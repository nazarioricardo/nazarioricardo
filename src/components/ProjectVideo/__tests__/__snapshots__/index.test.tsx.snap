// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectVideo matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="postVideo"
  >
    <div
      style="border-radius: 50px; overflow: hidden; width: 640px; height: 360px;"
    >
      <video
        controls=""
        preload="auto"
        src="https://test-video-url.com"
        style="width: 100%; height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
